tests/containers/buildah_podman.pm

```diff
@@ -24,29 +24,34 @@ use containers::urls 'get_suse_container_urls';
 use version_utils qw(get_os_release check_os_release);
 
 sub run {
-    my ($image_names, $stable_names) = get_suse_container_urls();
     my ($running_version, $sp, $host_distri) = get_os_release;
 
     install_buildah_when_needed($host_distri);
     install_podman_when_needed($host_distri);
     allow_selected_insecure_registries(runtime => 'podman');
     scc_apply_docker_image_credentials() if (get_var('SCC_DOCKER_IMAGE'));
 
-    for my $iname (@{$image_names}) {
-        record_info 'testing image', $iname;
-        test_container_image(image => $iname, runtime => 'buildah');
-        if (check_os_release('suse', 'PRETTY_NAME')) {
-            # Use container which it is created in test_container_image
-            # Buildah default name is conducted by <image-name>-working-container
-            my ($prefix_img_name) = $iname =~ /([^\/:]+)(:.+)?$/;
-            record_info $prefix_img_name;
-            test_opensuse_based_image(image => "${prefix_img_name}-working-container", runtime => 'buildah');
-            # Due to the steps from the test_opensuse_based_image previously,
-            # the image has been committed as refreshed
-            test_containered_app(runtime => 'podman',
-                buildah    => 1,
-                dockerfile => 'Dockerfile.suse',
-                base       => 'refreshed');
+    # We may test either one specific image VERSION or comma-separated CONTAINER_IMAGES
+    my $versions = get_var('CONTAINER_IMAGE_VERSIONS', get_required_var('VERSION'));
+    for my $version (split(/,/, $versions)) {
+        my ($untested_images, $released_images) = get_suse_container_urls($version);
+        my $images_to_test = check_var('CONTAINERS_UNTESTED_IMAGES', '1') ? $untested_images : $released_images;
+        for my $iname (@{$images_to_test}) {
+            record_info "IMAGE", "Testing image: $iname";
+            test_container_image(image => $iname, runtime => 'buildah');
+            if (check_os_release('suse', 'PRETTY_NAME')) {
+                # Use container which it is created in test_container_image
+                # Buildah default name is conducted by <image-name>-working-container
+                my ($prefix_img_name) = $iname =~ /([^\/:]+)(:.+)?$/;
+                record_info $prefix_img_name;
+                test_opensuse_based_image(image => "${prefix_img_name}-working-container", runtime => 'buildah');
+                # Due to the steps from the test_opensuse_based_image previously,
+                # the image has been committed as refreshed
+                test_containered_app(runtime => 'podman',
+                    buildah    => 1,
+                    dockerfile => 'Dockerfile.suse',
+                    base       => 'refreshed');
+            }
```

Comment:
This looks like a near-duplicate to the above. Looks like you are just fooling tools/detect_code_dups with the line `record_info $prefix_img_name;` in the middle. I strongly suggest to resolve the duplication first

Reply:
funny that we did basically same comment at the same time :) but you wrote yours in more demanding way. And obviously it is code duplication . But I always have strong opinion - I can not **require** from the person to fix **all** the issues which exist in the file which he/she modified. You can only require to not introduce new. If you look careful in the diff you will see that this duplication was there before this patch. So it would be nice if Jose will fix it but it is for not a strong requirement for this PR to be merged .

Reply:
removed

Reply:
@asmorodskyi Correct. I should have written that I already saw the duplication in the old code

Reply:
[poo#94000](https://progress.opensuse.org/issues/94000)

Reply:
cool, thx. solved :)